### PR TITLE
fix bind_invoice method

### PR DESCRIPTION
**Pull request by DinSmol**
see issue #1788

### CODE REVIEW

File: app/models/bank_transaction.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@ def bind_invoice(invoice_no, manual: false)
 
     invoice = Invoice.find_by(number: invoice_no)
     errors.add(:base, I18n.t('invoice_was_not_found')) unless invoice
```
Comment (yulgolem) [inline app/models/bank_transaction.rb:62]: Let's consider to move this check also to` #validate_invoice_data(invoice)` method.

**Review by yulgolem — APPROVED** [review]

**Review by yulgolem — CHANGES_REQUESTED** [review]
Let's add some tests to be sure.

**Review by yulgolem — APPROVED** [review]

**Comment by vohmar** [comment]
might be different issue, but maybe its and easy fix to do right here. the method paid error is indeed gone, but if a user creates a valid bank transaction that matches an unpaid invoice error is returned on pressing "bind manually"

undefined method `settings' for nil:NilClass

File: app/models/bank_transaction.rb [hunk 2]
```diff
@@ -102,7 +106,7 @@ def parsed_ref_number
   private
 
   def reset_pending_registrar_balance_reload
-    return unless registrar.settings['balance_auto_reload']
+    return unless registrar.present? && registrar.settings['balance_auto_reload']
```
Comment (yulgolem) [inline app/models/bank_transaction.rb:109]: Looks like we do not have this covered in tests, are we?

**Review by yulgolem — CHANGES_REQUESTED** [review]

**Comment by yulgolem** [comment]
I'd suggest to move a bind manually error `undefined method `settings' for nil:NilClass` to separate issue and complete/close this one - you know, just to eat the elephant one bite a time.

**Review by yulgolem — APPROVED** [review]

**Pull request merged**